test/integration/dockerapi_fetcher_test.go

```diff
@@ -39,3 +45,72 @@ func logAsJSON(t *testing.T, title string, data any) {
 	require.NoError(t, err)
 	t.Logf("%s: %s", title, string(b))
 }
+
+func TestDockerAPIFetcherAllMetricsPresent(t *testing.T) {
+	// Mocked container stats
+	mockedStats := types.StatsJSON{
+		Stats: types.Stats{
+			MemoryStats: types.MemoryStats{
+				Usage: 1024 * 1024 * 600, // 600 MB current memory usage, exceeding the Memory limit
+				Stats: map[string]uint64{
+					"file":         1024 * 1024 * 25, // 25 MB cache usage
+					"anon":         1024 * 1024 * 75, // 75 MB RSS usage
+					"kernel_stack": 1024 * 1024 * 5,  // 5 MB kernel stack usage
+					"slab":         1024 * 1024 * 5,  // 5 MB slab usage
+				},
+				Limit: 1024 * 1024 * 500, // 500 MB total memory limit
+			},
+		},
+	}
+
+	expectedMetrics := raw.Metrics{
+		Memory: raw.Memory{
+			UsageLimit:        1024 * 1024 * 500,  // 500 MB total memory limit
+			FuzzUsage:         1024 * 1024 * 600,  // 600 MB current memory usage
+			SwapUsage:         1024 * 1024 * 100,  // 100 MB swap usage (600 MB - 500 MB)
+			SwapLimit:         1024 * 1024 * 1000, // 1 GB total memory + swap limit
+			SoftLimit:         1024 * 1024 * 250,  // 250 MB memory reservation (soft limit)
+			Cache:             1024 * 1024 * 25,   // 25 MB cache usage
+			RSS:               1024 * 1024 * 75,   // 75 MB RSS usage
+			KernelMemoryUsage: 1024 * 1024 * 10,   // 10 MB kernel memory usage (kernel_stack + slab)
+		},
+	}
+
+	// Define mocked ContainerStats to include the ReadCloser
+	mockContainerStats := types.ContainerStats{
+		Body: mockContainerStatsReadCloser(mockedStats),
+	}
+
+	mockedContainerJSON := types.ContainerJSON{
+		ContainerJSONBase: &types.ContainerJSONBase{
+			ID: "test-container",
+			HostConfig: &container.HostConfig{
+				Resources: container.Resources{
+					Memory:            1024 * 1024 * 500,  // 500 MB
+					MemoryReservation: 1024 * 1024 * 250,  // 250 MB, for SoftLimit calculation
+					MemorySwap:        1024 * 1024 * 1500, // Total memory + swap
+				},
+			},
+		},
+	}
+
+	// Fetch the metrics for the mocked container
+	fetcherMock := NewDockerAPIFetcherMock(mockContainerStats, nil)
+	metrics, err := fetcherMock.Fetch(mockedContainerJSON)
```

Comment:
Did you consider defining this as an unit test instead? As I see it, we don't need to mock the fetcher (since we are calling the `Fetch` method directly) and we simply mock the underlying client.

It would be different from the approach taken for cgroups, but the cgroups use the fetcher through `biz.Processor` instead. If it useful, we could also add some integration tests but maybe we should use the the docker API directly. 

WDYT?